### PR TITLE
Add manager statistics format test

### PR DESCRIPTION
**Pull request by juliamagan**
|Related issue|
|-------------|
|    #3606     |

### Added

- Added `test_statistics_format`: checks the format of the statistics obtained through the API
---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @juliamagan  (Developer)  |    test_api/test_statistics       | [:green_circle:](https://ci.wazuh.info/job/Test_integration/34902/)⚫⚫ | [🟢 ](https://github.com/wazuh/wazuh-qa/files/10287813/R1-3606-centos-manager.tar.gz)[🟢 ](https://github.com/wazuh/wazuh-qa/files/10287814/R2-3606-centos-manager.tar.gz)[🟢 ](https://github.com/wazuh/wazuh-qa/files/10287815/R3-3606-centos-manager.tar.gz) |   CentOS      |   c01a09f    | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

File: tests/integration/test_api/test_statistics/test_statistics_format.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                                                t1_configuration_metadata)
+
+
+@pytest.mark.tier(level=0)
+@pytest.mark.filterwarnings('ignore::urllib3.exceptions.InsecureRequestWarning')
+@pytest.mark.parametrize('configuration, metadata', zip(t1_configurations, t1_configuration_metadata), ids=t1_case_ids)
+def test_enabled(configuration, metadata, load_wazuh_basic_configuration, set_wazuh_configuration,
+                 get_api_details, restart_wazuh_daemon_function):
+    """
+    description: Check if the stadidistics returned by the API have the expected format.
```
Comment (jmv74211) [inline tests/integration/test_api/test_statistics/test_statistics_format.py:34]:
```suggestion
    description: Check if the statistics returned by the API have the expected format.
```

File: tests/integration/test_api/test_statistics/data/test_cases/cases_statistics_format.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  configuration_parameters:
+    ENDPOINT: wazuh-analysisd
+  metadata:
+    endpoint: wazuh-analysisd
+- name: wazuh-db_endpoint
+  description: wazuh-db statistics endpoint
+  configuration_parameters:
+    ENDPOINT: wazuh-db
+  metadata:
+    endpoint: wazuh-db
```
Comment (jmv74211) [inline tests/integration/test_api/test_statistics/data/test_cases/cases_statistics_format.yaml:18]:
```suggestion
- name: remoted_endpoint
  description: remoted statistics endpoint
  configuration_parameters:
    ENDPOINT: wazuh-remoted
  metadata:
    endpoint: wazuh-remoted

- name: analysisd_endpoint
  description: analysisd statistics endpoint
  configuration_parameters:
    ENDPOINT: wazuh-analysisd
  metadata:
    endpoint: wazuh-analysisd

- name: wazuh-db_endpoint
  description: wazuh-db statistics endpoint
  configuration_parameters:
    ENDPOINT: wazuh-db
  metadata:
    endpoint: wazuh-db
```

File: tests/integration/test_api/test_statistics/test_statistics_format.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        - The `configuration_enabled` file provides the module configuration for this test.
+        - The `cases_enabled` file provides the test cases.
+    """
+    endpoint = metadata['endpoint']
+    complete_endpoint = f'/manager/daemons/stats?daemons_list={endpoint}'
+    api_details = get_api_details()
+    response = make_api_call(endpoint=complete_endpoint, headers=api_details['auth_headers'])
+    stats_schema = read_json_file(os.path.join(CONFIGURATIONS_PATH, f'{endpoint}_template.json'))
+
+    jsonschema.validate(instance=response.json(), schema=stats_schema)
```
Comment (jmv74211) [inline tests/integration/test_api/test_statistics/test_statistics_format.py:84]:
```suggestion

    # Check if the API statistics response data meets the expected schema. Raise an exception if not.
    jsonschema.validate(instance=response.json(), schema=stats_schema)
```

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/integration/test_api/test_statistics/test_statistics_format.py (hunk 1 above)
Comment (juliamagan) [reply tests/integration/test_api/test_statistics/test_statistics_format.py]: Done in 69c1db4

File: tests/integration/test_api/test_statistics/test_statistics_format.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        - Check that wazuh-analysisd daemon does not crash.
+
+    input_description:
+        - The `configuration_enabled` file provides the module configuration for this test.
+        - The `cases_enabled` file provides the test cases.
+    """
+    endpoint = metadata['endpoint']
+    complete_endpoint = f'/manager/daemons/stats?daemons_list={endpoint}'
+    api_details = get_api_details()
+    response = make_api_call(endpoint=complete_endpoint, headers=api_details['auth_headers'])
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/integration/test_api/test_statistics/test_statistics_format.py]: Done in 79022d0

File: tests/integration/test_api/test_statistics/test_statistics_format.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    stats_schema = read_json_file(os.path.join(CONFIGURATIONS_PATH, f'{endpoint}_template.json'))
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/integration/test_api/test_statistics/test_statistics_format.py]: Done in 79022d0

[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/integration/test_api/test_statistics/test_statistics_format.py]: Done in 79022d0

File: tests/integration/test_api/test_statistics/test_statistics_format.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    assertions:
+        - Check in the log that the EPS limitation has been activated.
+        - Check that wazuh-analysisd daemon does not crash.
+
+    input_description:
+        - The `configuration_enabled` file provides the module configuration for this test.
+        - The `cases_enabled` file provides the test cases.
+    """
+    endpoint = metadata['endpoint']
+    complete_endpoint = f'/manager/daemons/stats?daemons_list={endpoint}'
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply tests/integration/test_api/test_statistics/test_statistics_format.py]: Done in 79022d0

File: tests/integration/test_api/test_statistics/data/test_cases/cases_statistics_format.yaml (hunk 2 above)
Comment (juliamagan) [reply tests/integration/test_api/test_statistics/data/test_cases/cases_statistics_format.yaml]: Done in 6776f3a

File: tests/integration/test_api/test_statistics/test_statistics_format.py (hunk 3 above)
Comment (juliamagan) [reply tests/integration/test_api/test_statistics/test_statistics_format.py:84]: Done in b8e6c50

File: tests/integration/test_api/test_statistics/test_statistics_format.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+# Statistics format test configurations (t1)
+t1_configuration_parameters, t1_configuration_metadata, t1_case_ids = get_test_cases_data(t1_cases_path)
+t1_configurations = load_configuration_template(configurations_path, t1_configuration_parameters,
+                                                t1_configuration_metadata)
+
+
+@pytest.mark.tier(level=0)
+@pytest.mark.filterwarnings('ignore::urllib3.exceptions.InsecureRequestWarning')
+@pytest.mark.parametrize('configuration, metadata', zip(t1_configurations, t1_configuration_metadata), ids=t1_case_ids)
+def test_enabled(configuration, metadata, load_wazuh_basic_configuration, set_wazuh_configuration,
```
Comment (jmv74211) [inline tests/integration/test_api/test_statistics/test_statistics_format.py:31]: Change `test_enabled`

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/integration/test_api/test_statistics/test_statistics_format.py (hunk 7 above)
Comment (juliamagan) [reply tests/integration/test_api/test_statistics/test_statistics_format.py]: Done in 6bf2142

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**